<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="description_file" default="$(find mppi_mobile_panda)/resources/panda/panda.urdf"/>
    <!-- TF prefix -->
    <arg name="tf_prefix" default=""/>
    <!-- Load robot description -->
    <param name="robot_description" command="$(find xacro)/xacro $(arg description_file)"/>

    <!--  Run the test node-->
    <node name="panda_mobile_free" type="panda_mobile_free" pkg="mppi_mobile_panda" output="screen"
          launch-prefix="gdb -ex run --args">
        <rosparam command="load" file="$(find mppi_mobile_panda)/config/params.yaml"/>
    </node>

    <!-- Publish robot state (joints) as TF -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="100"/>
        <param name="use_tf_static"     value="true"/>
        <param name="tf_prefix"         value="$(arg tf_prefix)"/>
        <remap from="robot_description" to="robot_description"/>
        <remap from="joint_states"      to="/joint_states"/>
    </node>

    <!-- Open Rviz -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mppi_mobile_panda)/config/config.rviz" output="screen"/>
</launch>